README.md

```diff
@@ -4,17 +4,221 @@
 
-Please read our [Contribution Guide](https://simp.readthedocs.io/en/stable/contributors_guide/index.html).
+This module is optimally designed for use within a larger SIMP ecosystem, but it can be used independently:
+* When included within the SIMP ecosystem, security compliance settings will be
+  managed from the Puppet server.
+* If used independently, all SIMP-managed security subsystems will be disabled by
+  default and must be explicitly opted into by administrators.  Please review
+  [simp_options](https://github.com/simp/pupmod-simp-simp_options) for details.
+
+## Module Description
+
+This module provides a reasonably safe configuration of the main PAM stack
+focused on common security and compliance settings. Care has been taken to
+provide a significant set of switches and override mechanisms in order to
+provide for user flexibility.
+
+## Setup
+
+### Setup Requirements
+
+No special dependencies are required for core functionality of this module.
+
+You will need to download the
+[simp/oath](https://github.com/simp/pupmod-simp-simp_oath) if you want to use
+the **EXPERIMENTAL** OATH (TOTP/HOTP) support.
+
+### What ``pam`` Affects
+
+The ``pam`` module modifies various settings in the ``/etc/pam.d/`` and
+``/etc/security`` directories related to user logins via various authentication
+methods.
+
+## Usage
+
+### Basic Usage
+
+To set up PAM using a sane set of defaults, you can simply ``include`` the
+class as follows:
+
+```puppet
+include 'pam'
+```
+
+This will set up PAM with the following capabilities:
+
+* ``pwquality`` settings
+* ``faillock`` support with auto-unlocking
+* Password hash algorithm strengthening
+* Password history management
+* Automatic home directory creation
+* User TTY auditing (only ``root`` by default)
+* Overall default deny
+
+### Restricting System Logins (pam_access)
+
+To set up a 'default deny' policy for your system (local ``root`` logins are
+always allowed):
+
+```puppet
+include 'pam::access'
+```
+
+#### Managing System Access
+
+There are two methods for allowing users/groups into the system. The first is
+to use the ``pam::access::rule`` defined type.
+
+The parameters are named after their counterparts as defined in
+``access.conf(5)``.
+
+```puppet
+pam::access::rule { 'Allow Security Group from Anywhere':
+  users   => ['(security)'],
+  origins => ['ALL']
+}
+
+pam::access::rule { 'Allow Alice from Home':
+  users   => ['alice'],
+  origins => ['alice.home.net']
+}
+
+pam::access::rule { 'Allow Bob from Local':
+  users   => ['bob'],
+  origins => ['LOCAL'],
+  order   => 2000
+}
+
+pam::access::rule { 'Deny Bob from Remote':
+  users      => ['bob'],
+  origins    => ['ALL'],
+  permission => '-',
+  order      => 2001
+}
+```
+
+The second method is to define the access list as a ``Hash`` directly in Hiera:
+
+```yaml
+---
+pam::access::users:
+  defaults:
+    origins:
+      - ALL
+    permission: "+"
+    "(security)":
```

Comment:
Is this supposed to be empty? or are the users below items of it? If so then they should be indented

Reply:
It's empty